src/cljs/zetawar/game.cljs

```diff
@@ -1195,6 +1199,16 @@
       :game/self-repair (:self-repair settings-def)
       :game/move-through-friendly (:move-through-friendly settings-def)}]))
 
+(defn build-terrains-tx [db game-id terrain-type-id can-build-def]
+  (let [game (game-by-id db game-id)]
+    (into []
+          (map
+           (fn [unit-type-name]
+             {:db/id (db/next-temp-id)
+              :terrain-type/_can-build terrain-type-id
+              :terrain-build/unit-type (e (find-by db :unit-type/id (to-unit-type-id unit-type-name)))}))
```

Comment:
Terrain types should be related directly to unit types without an intermediate entity. Also, the find-by query will only work if there is a single ruleset loaded into the db. We want to be able to support multiple rulesets being loaded simultaneously.